doc/user-manual.md

```diff
@@ -0,0 +1,738 @@
+   - files (xlsx, json, csv)
+   - a string (a dhis2 period ?)
+   - a select between options
+
+Before running you'll get a small screen to enter them
+
+![](./user-manual-recipe-parameters.jpg)
+
+The parameters are defined currenly as json like this
+```json
+[
+    {
+        "id": "program",
+        "label": "Search for program",
+        "type": "dhis2",
+        "resourceName": "programs",
+        "default": "sample"
+    },
+    {
+        "id": "datalementTracker",
+        "label": "Search for tracker data element",
+        "type": "dhis2",
+        "resourceName": "dataElements",
+        "filter": "domainType:eq:TRACKER"
+    },
+    {
+        "id": "datalementAggregate",
+        "label": "Search for aggregate data element",
+        "type": "dhis2",
+        "resourceName": "dataElements",
+        "filter": "domainType:eq:AGGREGATE"
+    },
+    {
+        "id": "mode",
+        "label": "Select run mode",
+        "type": "select",
+        "default": "generateEmptyCsv",
+        "choices": [
+            [
+                "generateEmptyCsv",
+                "Generate an empty csv"
+            ],
+            [
+                "dryRun",
+                "Import from csv - Dry run"
+            ],
+            [
+                "import",
+                "Import from csv - import events"
+            ]
+        ]
+    },
+    {
+        "id": "file",
+        "label": "Pick csv with event values",
+        "type": "csv",
+        "helperText": "you can use 'Generate an empty csv' run mode to generate a template"
+    },
+     {
+        "id": "filexlsx",
+        "label": "Pick xlsx with the data",
+        "type": "xlsx",
+        "helperText": "you can use 'Generate an empty csv' run mode to generate a template"
+    }
+]
+```
+
+These parameters are then available in the recipe code
+
+```js
+const programId = parameters.program.id;
+
+const dryRun = parameters.mode =="dryRun";
+
+const events = parameters.file.data;
+```
+
+## 9. Recipe can return more than a table
+
+Some recipes needs to return more than a table.
+This is possible thanks to a small report defined as markdown.
+Based on [idyll-lang](https://idyll-lang.org/)
+
+![](./user-manual-idyll-render.jpg)
+
+The recipe register some datasets.
+
+```js
+report.register("stats_level2", stats);
+```
+
+The recipe has a report in markdown referencing these datasets, adding some contextual content.
+
+```markdown
+[PageOrientation orientation:"landscape" /]
+# Coordinates Coverage
+
+> Number of org units with coordinates
+> --------------------------------------------------------------------
+>                 Number of org units
+
+[PageBreak /]
+## At level 2
+[FlexBox]
+[OrgunitMap lines:stats_level2 /]
+[/FlexBox]
+[PageBreak /]
+
+```
+
+## 10. Accessing multiple dhis2 instances
+
+Some recipes can access 2 dhis2 instances to "compare" and "align" their metadata.
+This is also possible but requires credentials on both dhis2, and correct CORS settings.
```

Comment:
"be careful that this leaves the user/password accessible in the recipe" ? Or "make sure the password is passed as a param to avoid storing it in the recipe" ?